x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/conditional/CaseTests.java

```diff
@@ -36,166 +36,117 @@ public CaseTests(@Name("TestCase") Supplier<TestCaseSupplier.TestCase> testCaseS
      */
     @ParametersFactory
     public static Iterable<Object[]> parameters() {
-        // TODO this needs lots of stuff flipped to parameters
-        return parameterSuppliersFromTypedData(
-            List.of(new TestCaseSupplier("keyword", List.of(DataType.BOOLEAN, DataType.KEYWORD, DataType.KEYWORD), () -> {
+        List<TestCaseSupplier> suppliers = new ArrayList<>();
+        for (DataType type : List.of(
+            DataType.KEYWORD,
+            DataType.TEXT,
+            DataType.BOOLEAN,
+            DataType.DATETIME,
+            DataType.DATE_NANOS,
+            DataType.DOUBLE,
+            DataType.INTEGER,
+            DataType.LONG,
+            DataType.UNSIGNED_LONG,
+            DataType.IP,
+            DataType.VERSION,
+            DataType.CARTESIAN_POINT,
+            DataType.GEO_POINT,
+            DataType.CARTESIAN_SHAPE,
+            DataType.GEO_SHAPE
+        )) {
```

Comment:
Previously these were all written by hand, but we know a lot more about how to write these now.